babs/utils.py

```diff
@@ -1939,6 +1995,43 @@ def check_job_account(job_id_str, job_name, username_lowercase):
     jobs under qw, r, etc, or does not exist (not submitted);
     Also, the current username should be the same one as that used for job submission.
     """
+    if type_system == "sge":
+        return _check_job_account_sge(job_id_str, job_name, username_lowercase)
+    elif type_system == "slurm":
+        return _check_job_account_slurm(job_id_str, job_name, username_lowercase)
+
+
+def _check_job_account_slurm(job_id_str, job_name, username_lowercase):
+    """
+    get information for a finished job in Slurm by calling `sacct`
+    """
+    proc_qacct = subprocess.run(
+        ["sacct", "-u", username_lowercase,
+         "-j", job_id_str],
+        stdout=subprocess.PIPE
+    )
+
+    proc_qacct.check_returncode()
+    msg_l = proc_qacct.stdout.decode('utf-8').split("\n")
+    msg_head = msg_l[0].split()
+    if "State" not in msg_head or "JobID" not in msg_head or "JobName" not in msg_head:
+        return "sacct doesn't provide information about the job"
+
+    st_ind = msg_head.index("State")
+    jobid_ind = msg_head.index("JobID")
+    jobnm_ind = msg_head.index("JobName")
+    job_saact = msg_l[2].split() # the 2nd row should have the main job
+
+    if job_saact[jobid_ind] != job_id_str or job_saact[jobnm_ind] != job_name:
+        return "sacct doesn't have the info for the specific job or the format is different"
```

Comment:
The column of 'JobName' can be very narrow to hold the entire job name. I'm making the columns wider in PR #96.